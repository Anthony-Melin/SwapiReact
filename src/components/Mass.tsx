import { PropsWithChildren } from 'react'

type MassProps = PropsWithChildren<{ value: string }>

const Mass = ({ value, ...restProps }: MassProps) => {
    return (
        <p datatest-id="mass" {...restProps}>
            Mass: {value}
        </p>
    )
}

export default Mass
