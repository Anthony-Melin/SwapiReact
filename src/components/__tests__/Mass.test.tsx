import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import Mass from '../Mass.tsx'

describe('Mass', () => {
    beforeEach(() => {
        render(<Mass value="VALUE" />)
    })

    it('should match snapshot', () => {
        expect(screen.getByTestId('mass')).toMatchSnapshot()
    })

    it('should have text', () => {
        expect(screen.getByTestId('mass')).toHaveTextContent(
            'Mass: VALUE'
        )
    })
})
