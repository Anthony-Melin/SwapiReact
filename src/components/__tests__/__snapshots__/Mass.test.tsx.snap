// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mass > should match snapshot 1`] = `
<p
  data-testid="mass"
>
  Mass: 
  VALUE
</p>
`;
